### PR TITLE
xml: T6146: add utils and helper to provide priority data

### PR DESCRIPTION
**Pull request by jestabro**
## Change Summary

Replacement of priority.pl using xml cache data, and util for sorting config-sync sections.

Details and comparison in task.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6149

## Related PR(s)

https://github.com/vyos/vyos1x-config/pull/24

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/helpers/priority.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2023 VyOS maintainers and contributors
```
Comment (sever-sever) [inline src/helpers/priority.py]: 2024 :)

Comment (jestabro) [reply src/helpers/priority.py]: Thanks !

**Review by sever-sever — APPROVED** [review]

**Comment by sever-sever** [comment]
@jestabro Can we backport it to 1.4?

**Comment by jestabro** [comment]
As with some other recent work, I've set the task status as 'backport candidate' for a future backport, after 1.4.0; in this case, we _could_ consider it for 1.4.0 (requiring the companion PR https://github.com/vyos/vyos1x-config/pull/24), though I'm not sure it's that useful other than modernizing legacy stuff.

If we want it in 1.4.0, this and the companion PR should cause no harm.

**Pull request merged**